### ISSUE TITLE
Add time.Format support for permalinks

### ISSUE THREAD

**Issue by tgray** [gohugoio/hugo#4624]
It'd be nice to have the ability to have non-zero padded months and days for permalinks, mostly for compatibility with existing websites being imported to hugo.

I could probably code the change even though I know nothing about Go and have none of the build tools.  In the following function:
https://github.com/gohugoio/hugo/blob/20c9b6ec81171d1c586ea31d5d08b40b0edaffc6/hugolib/permalinks.go#L130

Something like the following case:

    case "monthnopad":
    return fmt.Sprintf("%d", int(p.Date.Month())), nil

Then again, I see a note for a more ambitious arbitrary strftime formatting capability.  That would work too.

**Comment by bep** [comment]
>Then again, I see a note for a more ambitious arbitrary strftime formatting capability. That would work too.

I don't see why we should support `strftime`... However, we should support some way of just using ´time.Format`, so you could put this string in `2006/1/2` ... **somehow**. That would be more consistent with how date formatting is in the rest of Hugo.

I'm not sure how to express that. 

Note that, the **somehow** must be explained before we can do this.

**Comment by jmooring** [comment]
Addressed a while ago with https://github.com/gohugoio/hugo/pull/6489
